Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidThrowingUnexpectedExceptionsAnalyzer.cs

```diff
@@ -0,0 +1,107 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	/// <summary>
+	/// Analyzer that checks exception are not thrown in locations in the code where they are not expected or cause issues.
+	/// </summary>
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidThrowingUnexpectedExceptionsAnalyzer : DiagnosticAnalyzer
+	{
+		private const string LocationsTitle = @"Avoid throwing exceptions from unexpected locations";
+		private const string LocationsMessageFormat = @"Avoid throwing exceptions from {0}, as this location is unexpected.";
+		private const string LocationsDescription = @"Avoid throwing exceptions from unexpected locations, like Finalizers, Dispose, Static Constructors, etc...";
+		private const string Category = Categories.Documentation;
+
+		private static readonly DiagnosticDescriptor LocationsRule = new(Helper.ToDiagnosticId(DiagnosticIds.AvoidExceptionsFromUnexpectedLocations), LocationsTitle, LocationsMessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: false, description: LocationsDescription);
+
+		[SuppressMessage("", "IDE0090", Justification = "When type is removed the literal type cannot be deduced")]
+		private static readonly Dictionary<string, string> specialMethods = new Dictionary<string, string>
```

Comment:
I noticed you don't use underscores for members. My teams have always done so, to easily distinguish from locals. (eg, when reviewing this, i spent a full 30 seconds trying to find the declaration of specialMethods in Analyze). The other arguably better approach is to use "this.".

Reply:
Will do from now on !